src/main/python/ayab/engine/options.py

```diff
@@ -18,19 +18,37 @@
 #    Andreas Müller, Christian Gerbrandt
 #    https://github.com/AllYarnsAreBeautiful/ayab-desktop
 
+from __future__ import annotations
+
 from enum import Enum
 
 from PySide6.QtCore import Qt, QCoreApplication
-from PySide6.QtWidgets import QWidget
+from PySide6.QtWidgets import QWidget,QComboBox
 from PySide6.QtGui import QPixmap
 
-from ayab.signal_sender import SignalSender
+from ..signal_sender import SignalSender
 from .options_gui import Ui_Options
 from .mode import Mode
-from ayab.machine import Machine
+from ..machine import Machine
+from .lowercase_e_rc import * 
+from .lowercase_e_reversed_rc import * 
+from typing import TYPE_CHECKING, Literal, Optional, TypedDict
+if TYPE_CHECKING:
+    from ..ayab import GuiMain
 
-import ayab.engine.lowercase_e_rc
-import ayab.engine.lowercase_e_reversed_rc
+OptionsTabDict = TypedDict('OptionsTabDict',{
+    "portname":str,
+    "machine": str, 
+    "mode": str,
+    "num_colors": int,
+    "start_row": int,
+    "inf_repeat": bool,
+    "start_needle": int,
+    "stop_needle": int,
+    "alignment": str,
+    "auto_mirror": bool,
+    "continuous_reporting": bool
+})
```

Comment:
I'm a big fan of these `TypedDict`s. They're getting used here and in Preferences, I think they go a long way towards making clear what's actually available!